libnext/src/parsers/nodes/endpoint-parser.ts

```diff
@@ -18,22 +19,25 @@ import { parseResponse } from "./response-parser";
  *
  * @param klass a class declaration
  */
-export function parseEndpoint(klass: ClassDeclaration): EndpointNode {
+export function parseEndpoint(
+  klass: ClassDeclaration
+): Locatable<EndpointNode> {
   const decorator = klass.getDecoratorOrThrow("endpoint");
-  const description = extractJsDocComment(klass);
+  const description = extractJsDocCommentLocatable(klass);
   const configuration = extractDecoratorFactoryConfiguration(decorator);
-  const tags = extractTagsProperty(configuration, "tags");
+  const tags = extractOptionalStringArrayPropertyValueLocatable(
+    configuration,
+    "tags"
+  );
   const method = extractHttpMethodProperty(configuration, "method");
-  const name = klass.getNameOrThrow();
-  const path = extractStringProperty(configuration, "path");
+  const name = {
+    value: klass.getNameOrThrow(),
+    location: klass.getSourceFile().getFilePath(),
+    line: klass.getNameNodeOrThrow().getStartLineNumber()
+  };
+  const path = extractStringPropertyValueLocatable(configuration, "path");
   const requestMethod = classMethodWithDecorator(klass, "request");
-  const request = requestMethod
-    ? parseRequest(requestMethod)
-    : {
-        headers: [],
-        pathParams: [],
-        queryParams: []
-      };
+  const request = requestMethod ? parseRequest(requestMethod) : undefined;
```

Comment:
FYI you can also do `requestMethod && parseRequest(requestMethod)` when the "else case" is `undefined`.